### DISCUSSION TITLE
v2.6.0

### DISCUSSION THREAD

**Discussion by gabegma** [ServiceNow/azimuth#522] [Announcements]
## What's Changed
* Add support for persistent ID and filtering on ID in exploration space by @gabegma in https://github.com/ServiceNow/azimuth/pull/301
* Bump express from 4.17.1 to 4.17.3 in /webapp by @dependabot in https://github.com/ServiceNow/azimuth/pull/328
* Fix similar and perturbed long utterances by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/330
* Remove warnings from GitHub actions by @gabegma in https://github.com/ServiceNow/azimuth/pull/326
* Improve-class-overlap-images by @gabegma in https://github.com/ServiceNow/azimuth/pull/329
* Fix webapp crash when no pipeline by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/333
* Update README.md by @gabegma in https://github.com/ServiceNow/azimuth/pull/337
* Fix sort by confidence or prediction without postprocessing by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/327
* Fix crash on Safari by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/339
* Update datasets by @gabegma in https://github.com/ServiceNow/azimuth/pull/309
* Log changes for 2.5.2 by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/342
* Update README.md by @gabegma in https://github.com/ServiceNow/azimuth/pull/343
* Bump decode-uri-component from 0.2.0 to 0.2.2 in /webapp by @dependabot in https://github.com/ServiceNow/azimuth/pull/323
* Bump loader-utils from 2.0.2 to 2.0.4 in /webapp by @dependabot in https://github.com/ServiceNow/azimuth/pull/289
* Bump wheel from 0.35.1 to 0.38.1 by @dependabot in https://github.com/ServiceNow/azimuth/pull/353
* Bump setuptools from 50.3.0 to 65.5.1 by @dependabot in https://github.com/ServiceNow/azimuth/pull/354
* Bump certifi from 2021.10.8 to 2022.12.7 by @dependabot in https://github.com/ServiceNow/azimuth/pull/334
* Bump joblib from 1.1.0 to 1.2.0 by @dependabot in https://github.com/ServiceNow/azimuth/pull/261
* Exclude fields from cache by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/350
* Clean up the last `useTheme()` by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/341
* Update @xmldom/xmldom by @gabegma in https://github.com/ServiceNow/azimuth/pull/356
* Added new fields in config page by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/282
* Add new start-up tasks by @gabegma in https://github.com/ServiceNow/azimuth/pull/332
* Fix app when no training set by @gabegma in https://github.com/ServiceNow/azimuth/pull/357
* Bump oauthlib from 3.2.0 to 3.2.1 by @dependabot in https://github.com/ServiceNow/azimuth/pull/249
* Bump protobuf from 3.19.4 to 3.19.5 by @dependabot in https://github.com/ServiceNow/azimuth/pull/254
* Save modules' metadata in cache by @gabegma in https://github.com/ServiceNow/azimuth/pull/352
* Use `mktemp` to store the OpenAPI spec by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/364
* Test `Description` by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/369
* Fix config scopes by @gabegma in https://github.com/ServiceNow/azimuth/pull/351
* Fix reference in the doc by @gabegma in https://github.com/ServiceNow/azimuth/pull/371
* Replace "simple" emojis shortcodes by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/372
* Support training set only by @gabegma in https://github.com/ServiceNow/azimuth/pull/361
* Add better support for csv files by @gabegma in https://github.com/ServiceNow/azimuth/pull/368
* Display config args as vertical list by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/367
* Improve config page display when values are null  by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/374
* Save config in cache when starting the app and when modifying config by @gabegma in https://github.com/ServiceNow/azimuth/pull/349
* Add units to config fields by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/370
* Fix Utterance Details page collapsing with extra long utterances by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/381
* Update citation and badges by @gabegma in https://github.com/ServiceNow/azimuth/pull/382
* Refactor Syntactic Warnings by @gabegma in https://github.com/ServiceNow/azimuth/pull/380
* Update poetry to 1.3.2 by @gabegma in https://github.com/ServiceNow/azimuth/pull/387
* Add persistent id to utterances by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/384
* Test various front-end components by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/224
* Refactor general warnings by @gabegma in https://github.com/ServiceNow/azimuth/pull/386
* Enforce mypy in pre-commit by @gabegma in https://github.com/ServiceNow/azimuth/pull/392
* Added default-config API route by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/389
* Update doc for syntax analysis by @gabegma in https://github.com/ServiceNow/azimuth/pull/393
* Remove generic tokenizer and support multiple languages for the word cloud. by @gabegma in https://github.com/ServiceNow/azimuth/pull/388
* Modify the minimum value to start from 1 for  all integer type config fields by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/394
* Enable `use_fast` for tokenizer by @gabegma in https://github.com/ServiceNow/azimuth/pull/401
* Improve optimistic update of `datasetInfo` by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/398
* Remove `model_contract` from `artifact_path` by @gabegma in https://github.com/ServiceNow/azimuth/pull/402
* Fix back end when updating the config by @gabegma in https://github.com/ServiceNow/azimuth/pull/395
* Add default config to config page UI by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/396
* Adapt dataset warnings so it works with one dataset split by @gabegma in https://github.com/ServiceNow/azimuth/pull/390
* Add a config scope for MetricsPerFilterModule by @gabegma in https://github.com/ServiceNow/azimuth/pull/399
* Fix config number fields by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/404
* Disable apply button while changes are being applied by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/405
* Add "Status" service API tag to trigger start up task on config update by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/397
* Add start-up task completion message before the application is loaded by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/403
* Improve router test by @gabegma in https://github.com/ServiceNow/azimuth/pull/407
* Inherit behavioral test config from AzimuthBaseSettings by @gabegma in https://github.com/ServiceNow/azimuth/pull/409
* Fix column name in dataset warnings by @gabegma in https://github.com/ServiceNow/azimuth/pull/410
* Support empty list for pipelines by @gabegma in https://github.com/ServiceNow/azimuth/pull/406
* Add a message while config is updated by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/412
* Add required_mod_options by @gabegma in https://github.com/ServiceNow/azimuth/pull/413
* Rename allowed_mod_options by @gabegma in https://github.com/ServiceNow/azimuth/pull/416
* Fix dashboard scroll by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/419
* Bump gitpython from 3.1.27 to 3.1.30 by @dependabot in https://github.com/ServiceNow/azimuth/pull/360
* Refactor flaky test by @gabegma in https://github.com/ServiceNow/azimuth/pull/417
* Fix flaky test by @gabegma in https://github.com/ServiceNow/azimuth/pull/422
* Bump oauthlib from 3.2.1 to 3.2.2 by @dependabot in https://github.com/ServiceNow/azimuth/pull/424
* Add all missing analyses config fields by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/426
* Refactor add_tags method by @gabegma in https://github.com/ServiceNow/azimuth/pull/423
* Export proposed actions by @gabegma in https://github.com/ServiceNow/azimuth/pull/408
* Refactor dataset manager by @gabegma in https://github.com/ServiceNow/azimuth/pull/427
* Compute syntax smart tags for utterances with multiple sentences. by @gabegma in https://github.com/ServiceNow/azimuth/pull/435
* Bump ipython from 8.2.0 to 8.10.0 by @dependabot in https://github.com/ServiceNow/azimuth/pull/436
* Adapt route to add proposed actions by @gabegma in https://github.com/ServiceNow/azimuth/pull/429
* Add language field to config page by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/428
* Correct the syntax section field name of config page by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/438
* Add truncation=True to HFPipelines by @Dref360 in https://github.com/ServiceNow/azimuth/pull/440
* Merge v2.5.2 by @gabegma in https://github.com/ServiceNow/azimuth/pull/448
* Create v2.5.3 release by @gabegma in https://github.com/ServiceNow/azimuth/pull/449
* Add short and long utterances on token count plot by @gabegma in https://github.com/ServiceNow/azimuth/pull/442
* Add persistent id to front end by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/385
* Fix generated types for `Union`s in `Dict`s by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/455
* Change the outcome per threshold bar chart to area chart by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/444
* Fix `patch_utterances` being a `POST` by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/457
* Bump isort in pre-commit by @gabegma in https://github.com/ServiceNow/azimuth/pull/458
* Save task results when loading module cache by @gabegma in https://github.com/ServiceNow/azimuth/pull/459
* Support ignoring missing persistent ids when importing proposed actions by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/461
* Add export utterances, export proposed actions to utterance table by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/445
* Refactor get_utterances by @gabegma in https://github.com/ServiceNow/azimuth/pull/452
* Update `poetry.lock` by @gabegma in https://github.com/ServiceNow/azimuth/pull/463
* Migrate dev dependencies to new poetry group by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/462
* Remove `/admin` route prefix by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/466
* Test for performance analysis comparison table by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/420
* Fix front-end tests by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/468
* Add fix to browser history after navigating to exploration page with no pipeline by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/464
* Add Outcome option to dropdown in Smart Tag Analysis by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/456
* Remove v1 folder structure by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/467
* Clean up pipeline vs model and previous PR by @gabegma in https://github.com/ServiceNow/azimuth/pull/450
* UI test script for class overlap table by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/441
* Link from confusion matrix to utterance table by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/470
* Fix negative query params by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/471
* Bump json5 from 1.0.1 to 1.0.2 in /webapp by @dependabot in https://github.com/ServiceNow/azimuth/pull/359
* Update tensorflow by @gabegma in https://github.com/ServiceNow/azimuth/pull/355
* Bump dns-packet from 5.3.1 to 5.4.0 in /webapp by @dependabot in https://github.com/ServiceNow/azimuth/pull/473
* Bump werkzeug from 2.1.0 to 2.2.3 by @dependabot in https://github.com/ServiceNow/azimuth/pull/443
* Validate config min/max values by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/474
* Clean up docker image by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/316
* Load config history by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/472
* Bump jupyter-core from 4.9.2 to 4.11.2 by @dependabot in https://github.com/ServiceNow/azimuth/pull/477
* Bump torch from 1.9.0 to 1.13.1 by @dependabot in https://github.com/ServiceNow/azimuth/pull/379
* Update tensorflow to 2.11.0 by @gabegma in https://github.com/ServiceNow/azimuth/pull/476
* Import proposed actions to utterance table by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/465
* Support dataset-only smart tag analysis by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/478
* Fix outcome per threshold plot overflow by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/485
* Add modal to config page by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/437
* Update documentation for 2.6.0 by @gabegma in https://github.com/ServiceNow/azimuth/pull/484
* Fix unshrinking labels by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/489
* Hide TF info messages in logs by @gabegma in https://github.com/ServiceNow/azimuth/pull/492
* Create `AzimuthConfigHistory` class by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/493
* Add "created-on" in config history by @gabegma in https://github.com/ServiceNow/azimuth/pull/491
* Add faiss to startup by @gabegma in https://github.com/ServiceNow/azimuth/pull/495
* Remove artifact_path from project hash by @gabegma in https://github.com/ServiceNow/azimuth/pull/497
* Allow user to change dask cluster from route by @gabegma in https://github.com/ServiceNow/azimuth/pull/500
* Make the Swagger UI work through the front-end proxy by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/502
* Fix config page checkbox fields click event by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/496
* Simplify Makefile by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/503
* Fix errors after applying config changes that require startup tasks by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/501
* Make API errors uniform by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/480
* Avoid parsing whole config prematurely by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/504
* Clean up config vs task_manager.config by @gabegma in https://github.com/ServiceNow/azimuth/pull/499
* Revert "Simplify Makefile (#503)" by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/508
* Support serving the front end in HTTPS by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/507
* Improve config modal by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/487
* Fix overwriting the partial config on changing the language field by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/509
* Save config only when different from last config in history by @gabegma in https://github.com/ServiceNow/azimuth/pull/506
* Separate Docker build in two GitHub actions by @gabegma in https://github.com/ServiceNow/azimuth/pull/511
* Fix GPU release by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/514
* Make config UI string fields editable by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/482
* Add common fields configuration section and persistentId under the project section by @nandhinibsn in https://github.com/ServiceNow/azimuth/pull/505
* Update numpy to 1.23.5 by @gabegma in https://github.com/ServiceNow/azimuth/pull/516
* Enable debug option by @gabegma in https://github.com/ServiceNow/azimuth/pull/517
* Support editing lists, args, and kwargs by @JosephMarinier in https://github.com/ServiceNow/azimuth/pull/512
* Tackle dask issues by @gabegma in https://github.com/ServiceNow/azimuth/pull/518
* Document known issue by @gabegma in https://github.com/ServiceNow/azimuth/pull/520
* Release 2.6.0 by @gabegma in https://github.com/ServiceNow/azimuth/pull/519

**Full Changelog**: https://github.com/ServiceNow/azimuth/compare/v2.5.3...v2.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/ServiceNow/azimuth/releases/tag/v2.6.0'>v2.6.0</a>.</em>
